packages/dds/map/src/test/directory.spec.ts

```diff
@@ -142,6 +143,51 @@ describe("Directory", () => {
                 assert.equal(clearExpected, false, "missing clearExpected event");
             });
 
+            it("Should fire dispose event correctly", () => {
+                let valueChangedExpected: boolean = true;
+                let previousValue: any;
+
+                directory.on("valueChanged", (changed, local, target) => {
+                    assert.equal(valueChangedExpected, true, "valueChange event not expected");
+                    valueChangedExpected = false;
+
+                    assert.equal(changed.key, "dwayne", "key should match");
+                    assert.equal(changed.previousValue, previousValue, "previous value should match");
+                    assert.equal(changed.path, "/rock", "absolute path should match");
+
+                    assert.equal(local, true, "local should be true for local action for valueChanged event");
+                    assert.equal(target, directory, "target should be the directory for valueChanged event");
+                });
+
+                // Test dispose on subdirectory delete
+                let subDirectoryDisposed = false;
+                const subDirectory = directory.createSubDirectory("rock");
+                subDirectory.on("disposed", (value: IDirectory) => {
+                    subDirectoryDisposed = true;
+                    assert.equal(value.disposed, true, "sub directory not deleted");
+                });
+                // Should fire dispose event.
+                directory.deleteSubDirectory("rock");
+                assert.equal(subDirectoryDisposed, true, "sub directory not disposed!!");
+
+                // Should be able to work on new directory with same name.
+                previousValue = undefined;
+                valueChangedExpected = true;
+                const newSubDirectory = directory.createSubDirectory("rock");
+                newSubDirectory.set("dwayne", "johnson");
+                assert.equal(valueChangedExpected, false, "missing valueChangedExpected event");
+
+                // Usage Error on accessing disposed directory.
+                let usageErrorExpected = true;
+                try {
+                    subDirectory.set("throw", "error");
+                } catch (error) {
+                    assert.strictEqual(error.errorType, "usageError", "Should throw usage error");
+                    usageErrorExpected = false;
+                }
+                assert(usageErrorExpected === false, "Should throw usage error");
```

Comment:
You can get rid of usageErrorExpected altogether by addign assert(false) after line 183